Tools/CMake/Conan.cmake

```
@@ -23,55 +23,93 @@ if(USE_CONAN)
 
   message(STATUS "  ${CMAKE_BUILD_TYPE}")
   set(CONAN_COMPILER_RUNTIME "dynamic")
-  set(CONAN_RESULT_FILE "conanbuild.bat") #for windows
 
+
+  # We use our own recipe with some patches to cook up a functional version of freexl, so get the recipe:
+  message(STATUS "Cloning private freexl dependency")
+  set(FREEXL_VERSION "2.0.99.cci.20250526")
+  FetchContent_Declare(
+    freexl
+    GIT_REPOSITORY   https://github.com/jasp-stats/conan-recipes.git
+    GIT_TAG          e02cefc5a37684749a20d31500fc54468383e30a
+  )
+  FetchContent_MakeAvailable(freexl)
+
+  # Configure Conan for windows
   if(WIN32)
+    set(CONAN_RESULT_FILE "conanbuild.bat") #for windows
 
     message(STATUS "  ${CONAN_COMPILER_RUNTIME}")
-
+    
     execute_process(
       COMMAND_ECHO STDOUT
       WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
       COMMAND
-      conan install ${CONAN_FILE_PATH} --output-folder=${CMAKE_BINARY_DIR}/conan_build
+      conan install ${CONAN_FILE_PATH} --output-folder=${CMAKE_BINARY_DIR}/_conan_build
       -s build_type=${CMAKE_BUILD_TYPE}
       -c tools.cmake.cmaketoolchain:generator=${CMAKE_GENERATOR}
       -s compiler.runtime=${CONAN_COMPILER_RUNTIME} --build=missing)
 
+
+      if(NOT freexl_FOUND)
+        if(freexl_POPULATED)
+            message(STATUS "Compiling freexl dependency ${freexl_SOURCE_DIR}")
+            execute_process(
+                COMMAND_ECHO STDOUT
+                WORKING_DIRECTORY ${freexl_SOURCE_DIR}/freexl
+                COMMAND
+                conan create ${freexl_SOURCE_DIR}/freexl --version=${FREEXL_VERSION}
+                -s build_type=${CMAKE_BUILD_TYPE}
+                -c tools.cmake.cmaketoolchain:generator=${CMAKE_GENERATOR}
+                -s compiler.runtime=${CONAN_COMPILER_RUNTIME} --build=missing
+                --test-missing
+            )
+        else()
+          message(CHECK_FAIL "build freexl failed")
+        endif()
+    endif()
+  
+      # configure conan for apple
   elseif(APPLE)
 
     set(CONAN_RESULT_FILE "conanbuild.sh")
-
-    if(CROSS_COMPILING)
-
-      execute_process(
+    
+    execute_process(
         COMMAND_ECHO STDOUT
         WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-        COMMAND
-          zsh -l -c "conan install ${CONAN_FILE_PATH} -s build_type=${CMAKE_BUILD_TYPE} -s os.version=${CMAKE_OSX_DEPLOYMENT_TARGET} --build=missing -of ${CMAKE_BINARY_DIR}/conan_build")
+        COMMAND zsh -c -l "export CC=\"\"; export CCX=\"\"; conan install ${CONAN_FILE_PATH} -s build_type=${CMAKE_BUILD_TYPE} -s os.version=${CMAKE_OSX_DEPLOYMENT_TARGET} --build=missing -of ${CMAKE_BINARY_DIR}/_conan_build")
     
-    else()
+    execute_process(
+      COMMAND_ECHO STDOUT
+      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+      COMMAND zsh -c -l "export CC=\"\"; export CCX=\"\"; conan create ${freexl_SOURCE_DIR}/freexl --version=${FREEXL_VERSION} -s build_type=${CMAKE_BUILD_TYPE} -s os.version=${CMAKE_OSX_DEPLOYMENT_TARGET} --build=missing")        
```

```suggestion
      COMMAND zsh -c -l "export CC=\"\"; export CCX=\"\"; conan create ${freexl_SOURCE_DIR}/freexl --version=${FREEXL_VERSION} -s build_type=${CMAKE_BUILD_TYPE} -s os.version=${CMAKE_OSX_DEPLOYMENT_TARGET} --build=missing --test-missing")        
```

It's better to have this if build freexl locally otherwise will failed sometimes cause it will try to build an example within conan env.

---

Ill test it